sympy/simplify/simplify.py

```diff
@@ -2540,8 +2541,8 @@ def recurse(arg, **kwargs):
 
     expr = sympify(expr)
 
-    if not isinstance(expr, Basic) or expr.is_Atom or expr in (
-            exp_polar(0), exp_polar(1)):
+    if (not isinstance(expr, Basic) or isinstance(expr, MatrixSymbol) or (
+            expr.is_Atom or expr in (exp_polar(0), exp_polar(1)))):
```

Comment:
I think the logic in the fix is still reasonable. For power simplification, if we encounter ```MatrixSymbol```, we can obviously return the symbol itself without further processing.